codec-redis/src/main/java/io/netty/handler/codec/redis/RedisDecoder.java

```diff
@@ -0,0 +1,232 @@
+        } else if (bulkStringLength == 0L) {
+            // $0\r\n <here> \r\n
+            if (in.readableBytes() < CRLF_LENGTH) {
+                return false;
+            }
+            in.skipBytes(CRLF_LENGTH);
+            addMessageToOut(out, RedisMessageFactory.emptyBulkString());
+        } else if (bulkStringLength > 0L) {
+            // ${bulkStringLength}\r\n <here> {data...}\r\n
+            if (in.readableBytes() < bulkStringLength + CRLF_LENGTH) {
+                return false;
+            }
+            ByteBuf content = in.readSlice((int) bulkStringLength);
+            in.skipBytes(CRLF_LENGTH);
+            addMessageToOut(out, RedisMessageFactory.createBulkString(content.retain()));
+        } else {
+            throw new CodecException("bad bulkStringLength: " + bulkStringLength);
+        }
+        return true;
+    }
+
+    private void addMessageToOut(List<Object> out, RedisMessage msg) {
+        out.add(msg);
+        resetDecoder();
+    }
+
+    private RedisMessage newInlineRedisMessage(RedisMessageType messageType, ByteBuf bytes) {
+        switch (messageType) {
+        case SIMPLE_STRING:
+            return RedisMessageFactory.createSimpleString(bytes);
+        case ERROR:
+            return RedisMessageFactory.createError(bytes);
+        case INTEGER:
+            return RedisMessageFactory.createInteger(parseRedisNumber(bytes));
+        default:
+            throw new CodecException("bad type: " + messageType);
+        }
+    }
+
+    private static ByteBuf readLine(ByteBuf in) {
+        final int lfIndex = in.forEachByte(ByteProcessor.FIND_LF);
+        if (lfIndex < 0) {
+            return null;
+        }
+        final int length = lfIndex - in.readerIndex() - 1; // `-1` is for CR
+        return readBytes(in, length);
+    }
+
+    private static ByteBuf readBytes(ByteBuf in, int length) {
+        ByteBuf data = in.readSlice(length);
+        in.skipBytes(CRLF_LENGTH); // skip CRLF
+        return data;
+    }
+
+    private long parseRedisNumber(ByteBuf byteBuf) {
+        toLongProcessor.result = 0;
+        toLongProcessor.first = true;
+        byteBuf.forEachByte(toLongProcessor);
+        if (!toLongProcessor.negative) {
+            return -toLongProcessor.result;
+        }
+        return toLongProcessor.result;
+    }
+
+    private static final class ToLongProcessor implements ByteProcessor {
+        private long result;
+        private boolean negative;
+        private boolean first;
+
+        @Override
+        public boolean process(byte value) throws Exception {
+            if (first) {
+                first = false;
+
+                if (value == '-') {
+                    negative = true;
+                } else {
+                    negative = false;
+                    int digit = value - '0';
+                    result = result * 10 - digit;
+                }
+                return true;
+            }
+
+            int digit = value - '0';
```

Comment:
`if (value < '0' || value > '9')` should be equivalent to `if (value < 48 || value > 57)` no?